### PR TITLE
explorer: add ANS domains integration

### PR DESCRIPTION
**Pull request by tonivlad**
#### Summary of Changes
Added ANS (Alternative Name Service) Protocol domains to the Explorer
- search by domain
- list owned domains

ANS enables any project on Solana to have their own TLD ( _.anything_ )

currently only **.poor** domains are enabled in the protocol 

we are in talks with **MonkeDAO**, **ThugDao**, **Frakt** and a few other projects to enable **.monke**, **.thug**, **.frakt**, etc

Fixes #

####  Previews 

1. Owned domains
explorer_owned

2. Search by domain
explore_solana

### CODE REVIEW

**Comment by ngundotra** [comment]
@tonivlad can you please open an issue, tagged with `explorer` that describes what `onsol` domains are, how to register them, etc?

**Comment by hydrogenbond007** [comment]
🫡

**Comment by tonivlad** [comment]
@ngundotra I will open an issue tomorrow explaining how everything works

**Comment by tonivlad** [comment]
@ngundotra 

- we published the IDL for the program you requested on Twitter
- here's link to the tldparser package: [https://github.com/onsol-labs/tld-parser](https://github.com/onsol-labs/tld-parser)
- I created an issue about this
#29309

ps: it's a bit wordy but I wanted to give more context

**Comment by tonivlad** [comment]
@aeyakovenko 

The process to create a new TLD is first come first serve but currently involves talking to us

we are trying to filter the first projects to make sure they are they are projects that have stood the test of time and have a long term plan

we're doing this also because we want to see what other features they require

please check out #29309, I added some context there

**Comment by solserer-labs** [comment]
This is really dope! I think this is an extremely valuable addition to Solana and will set us apart and make us first movers in the domain name space instead of being followers. Right now there is only "native" token extensions such as .eth .sol in the space. These are nice but I think allowing folks/projects/communities to personalize their own TLDs can add a lot of value for everyone. In the future I fully expect ANS to be integrated into browsers such as brave and some of these TLDs to become as prominent as something like .io or .ai in the legacy TLD space.

**Comment by cloakd** [comment]
Nice PR, would be good to get this in & ensure parity across other explorers for the naming house so users have a consistent UX when utilizing the Naming service across dApp's  - Both BlokHost & AlphaBatem are also in the process of supporting ANS on their explorers currently

**Comment by ngundotra** [comment]
We will come back to this in the new year!

**Comment by josh0x** [comment]
Good idea!

**Comment by stegaBOB** [comment]
Are the ANS programs open source? I briefly looked but couldn't find them anywhere. 

I think the expiration feature is cool, but it could also be implemented around the SNS program too (since the parent registry can transfer child domains). As far as I know, Bonfida doesn't actually control the program anymore. It is in the SPL for anyone to use. They just have their own registry account (which anyone can make) and the .sol and stuff are registered under their account. Having more names is cool, but adding another implementation of a primitive that already exists seems like it will make dealing with names quite a bit more cumbersome in the future.

**Comment by ngundotra** [comment]
We're working on generalizing Contract homepages so ecosystem teams can add information about how to use their smart contracts.

https://github.com/solana-labs/solana/issues/29556 lays out what we think is acceptable. What would you help you reformat ANS into this sort of contract? We are open to discussing changes on that issue :)
